### PR TITLE
Fix designate/TLS integration

### PR DESCRIPTION
**Pull request by Alex-Welsh**
When designate is deployed with the new internal TLS configuration, the first service deploy will fail if the barbican role IDs are undefined. This change assumes that secrets.yml already exists and replaces the contents rather than creating it from scratch.

### CODE REVIEW

**Comment by Alex-Welsh** [comment]
See also https://github.com/stackhpc/stackhpc-kayobe-config/pull/566

**Comment by jackhodgkiss** [comment]
We will need to make sure uuidgen is present on the system as it wasn't present in the Ubuntu 20.04 image I used.

**Comment by Alex-Welsh** [comment]
> We will need to make sure uuidgen is present on the system as it wasn't present in the Ubuntu 20.04 image I used.

I thought for a while about adding it as either a requirement for SKC or a play in the TLS playbooks but I think it's probably best here. It's a useful tool but isn't strictly necessary

**Review by jackhodgkiss — APPROVED** [review]
LGTM.

If we start to need more deployment dependencies I think we would want to have a playbook to manage them.

File: templates/deploy-openstack.tpl [hunk 1]
```diff
@@ -83,16 +89,11 @@ sed -i 's/# kolla_enable_tls_internal: true/kolla_enable_tls_internal: true/g' $
 cat $KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/kolla/globals-tls-config.yml >> $KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/kolla/globals.yml
 
 # Create vault configuration for barbican
-cat << EOF >> $KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/secrets.yml
----
-secrets_barbican_approle_secret_id: $(uuidgen)
-EOF
+sed -i "s/secret_id:.*/secret_id: $(uuidgen)/g" KAYOBE_CONFIG_PATH/environments/$KAYOBE_ENVIRONMENT/secrets.yml
```
Comment (markgoddard) [inline templates/deploy-openstack.tpl:92]: Isn't the secrets.yml file encrypted at this point?

Comment (Alex-Welsh) [reply templates/deploy-openstack.tpl:92]: So it is. I thought I had added a decrypt line before it but apparently not. I'll push a fix

Comment (Alex-Welsh) [reply templates/deploy-openstack.tpl:92]: https://github.com/stackhpc/terraform-kayobe-multinode/pull/30

**Pull request merged**